shell-control/agent/backend/agent_tools.py

```
@@ -1,59 +1,24 @@
-                }
-
-    def get_by_name(self, name: str) -> Tuple[Optional[Callable], Optional[Callable]]:
-        tool = self.toolset.get(name)
-        if not tool: 
-            return None, None
-
-        function = tool.get("function")
-        formatter_function = tool.get("formatter_function", None)
-        return function, formatter_function
+class AgentTools(ToolsBase):
+    def __init__(self, settings: dict = {}):
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)